server/internal/entity_internal/lightning.go

```diff
@@ -0,0 +1,61 @@
+package entity_internal
+
+import (
+	"github.com/df-mc/dragonfly/server/block"
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/entity"
+	"github.com/df-mc/dragonfly/server/entity/physics"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/go-gl/mathgl/mgl64"
+	"math"
+	"math/rand"
+)
+
+//go:linkname world_performThunder github.com/df-mc/dragonfly/server/world.performThunder
+//noinspection ALL
+var world_performThunder func(w *world.World, pos world.ChunkPos)
+
+func init() {
+	world_performThunder = func(w *world.World, pos world.ChunkPos) {
```

Comment:
Would pass the lcg rand instance to this function so you can get rid of w.ThunderLCG().

Reply:
How would that work if we have a rand instance per world? We can also just do a global one.

Reply:
Just have the caller pass its rand instance to this function.